src/sharding/ShardingManager.js

```diff
@@ -101,6 +107,32 @@ class ShardingManager extends EventEmitter {
     return shard;
   }
 
+  /**
+   * Kills all Shards without them respawning.
+   */
+  killAll() {
+    const _respawn = this.respawn;
+    this.respawn = false;
+    for (const shard of this.shards.values()) shard.kill();
+    this.respawn = _respawn;
```

Comment:
manipulating `this.respawn` to be false when killing the shard is useless, as `Shard#kill` ensures that the respawn will not occur.

Reply:
Fixing your correct, I forgot to make the correct edits too that.